ffi/main.go

```diff
@@ -133,6 +137,87 @@ func UpdateCheck(socket, fromArchivePath string) {
 	}()
 }
 
+//export GetLatestLicense
+func GetLatestLicense(licenseData string) *C.char {
+	kotsscheme.AddToScheme(scheme.Scheme)
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	obj, _, err := decode([]byte(licenseData), nil, nil)
+	if err != nil {
+		fmt.Printf("failed to decode license data: %s\n", err.Error())
+		return C.CString("")
+	}
+	license := obj.(*kotsv1beta1.License)
+
+	u, err := url.Parse(license.Spec.Endpoint)
+	if err != nil {
+		fmt.Printf("failed to parse endpoint from license: %s\n", err.Error())
+		return C.CString("")
+	}
+
+	hostname := u.Hostname()
+	if u.Port() != "" {
+		hostname = fmt.Sprintf("%s:%s", u.Hostname(), u.Port())
+	}
+
+	url := fmt.Sprintf("%s://%s/release/%s/license", u.Scheme, hostname, license.Spec.AppSlug)
```

Comment:
actually, you can just assume license endpoint format is what should be used, so you don't need to generate `hostname`